x/datapool/keeper/pool_distribution.go

```diff
@@ -177,42 +204,22 @@ func (k Keeper) getAvailablePoolCoinAmount(ctx sdk.Context, pool *types.Pool) (*
 	}
 }
 
-func (k Keeper) executeRevenueDistribution(ctx sdk.Context, pool *types.Pool) error {
-	availablePoolCoinAmount, err := k.getAvailablePoolCoinAmount(ctx, pool)
-	if err != nil {
-		return err
-	}
-
-	// if there is no coin available, proceed no further.
-	if availablePoolCoinAmount.Equal(sdk.NewInt(0)) {
-		return nil
-	}
-
-	// calculate the revenue to be sent to each seller
-	eachDistributionAmount := k.getEachDistributionAmount(pool)
-
+func (k Keeper) distributionToEachSeller(ctx sdk.Context, pool *types.Pool, eachSellerDistributionAmount sdk.Int, availablePoolCoinAmount *sdk.Int) error {
```

Comment:
The `availablePoolCoinAmount` here is required?
It is just updated after distribution of each sales history, but not used for anything I think. wdyt?

Reply:
After calculating the amount paid, we are updating the history.
```go
 
    paymentAmount := calculatePaymentAmountToSeller(history, eachSellerDistributionAmount)

    ...

    *availablePoolCoinAmount = availablePoolCoinAmount.Sub(paymentAmount)

    k.SetSalesHistory(ctx, history)
```

If you do not receive availablePoolCoinAmount, you must recalculate paymentAmount elsewhere.

Reply:
Now I understand.
Do not use this variable! T_T
Thank you for your detailed review.